### ISSUE TITLE
`yay -Qu` fails with error

### ISSUE THREAD

**Issue by h3fang** [Jguer/yay#1158]
#### Affected Version
yay v9.4.3 - libalpm v12.0.1

#### Issue
`yay -Qu` fails with error

#### Steps to reproduce
1. run `yay -Qu`

#### Output
`yay -Qu`

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/Jguer/yay/v9/pkg/stringset.StringSet.Set(...)
	pkg/stringset/stringset.go:27
main.upList(0xc000101d70, 0x391, 0x400, 0xc0002b8200, 0x10, 0x10, 0xc0002ec000, 0x391, 0x400)
	upgrade.go:133 +0x1095
main.printUpdateList(0xc000082300, 0xc000101e58, 0x2)
	print.go:404 +0x12b
main.handleQuery(0xa17ab1, 0x1)
	cmd.go:181 +0x7c
main.handleCmd(0x0, 0x0)
	cmd.go:155 +0x1e9
main.main()
	main.go:231 +0xeb
```

`yay -Pg`

```
{
	"aururl": "https://aur.archlinux.org",
	"buildDir": "/home/enigma/.cache/yay",
	"absdir": "/home/enigma/.cache/yay/abs",
	"editor": "",
	"editorflags": "",
	"makepkgbin": "makepkg",
	"makepkgconf": "",
	"pacmanbin": "pacman",
	"pacmanconf": "/etc/pacman.conf",
	"redownload": "no",
	"rebuild": "no",
	"batchinstall": false,
	"answerclean": "",
	"answerdiff": "",
	"answeredit": "",
	"answerupgrade": "",
	"gitbin": "git",
	"gpgbin": "gpg",
	"gpgflags": "",
	"mflags": "",
	"sortby": "votes",
	"searchby": "name-desc",
	"gitflags": "",
	"removemake": "ask",
	"sudobin": "sudo",
	"sudoflags": "",
	"requestsplitn": 150,
	"sortmode": 0,
	"completionrefreshtime": 7,
	"sudoloop": false,
	"timeupdate": false,
	"devel": false,
	"cleanAfter": false,
	"provides": true,
	"pgpfetch": true,
	"upgrademenu": true,
	"cleanmenu": true,
	"diffmenu": true,
	"editmenu": false,
	"combinedupgrade": false,
	"useask": false
}
```

**Comment by gtnuux** [comment]
I have exactly the same problem after upgrading to 9.4.3.  I got as far as figuring out that it is triggered by having  an IgnorePkg entry in /etc/pacman.conf  for an AUR package.  If I comment out my 'IgnorePkg=google-chrome' entry, then yay -Qu works correctly.  IgnorePkg entries for regular "repo" packages do not seem to trigger this bug.

Thanks,
-Jeff

**Comment by h3fang** [comment]
> I have exactly the same problem after upgrading to 9.4.3. I got as far as figuring out that it is triggered by having an IgnorePkg entry in /etc/pacman.conf for an AUR package. If I comment out my 'IgnorePkg=google-chrome' entry, then yay -Qu works correctly. IgnorePkg entries for regular "repo" packages do not seem to trigger this bug.
> 
> Thanks,
> -Jeff

Similar to your pacman config. I ignored a group of packages `IgnoreGroup = modified`. It contains official repo packages, AUR packages, and personal packages.

**Comment by Jguer** [comment]
Ok guys, thanks for the help triaging this, I've been to reproduce it and I'm cranking out a fix

**Comment by Jguer** [comment]
Fixed in 9.4.4

**Comment by gtnuux** [comment]
WOW! Thanks for fixing this with lightning speed!
I just built yay 9.4.4 from the AUR (using yay 9.4.3), and can verify that it no longer crashes.
-Jeff
